File: Multi_Language_CodeExamples/haskell/list_operations.hs
-- Haskell List Operations

numbers :: [Int]
numbers = [1, 2, 3, 4, 5]

main :: IO ()
main = do
    let doubled = map (* 2) numbers
    putStrLn $ "Doubled: " ++ show doubled
    
    let sumTotal = sum numbers
    putStrLn $ "Sum: " ++ show sumTotal

File: Multi_Language_CodeExamples/erlang/list_operations.erl
-module(list_ops).
-export([main/0]).

main() ->
    Numbers = [1, 2, 3, 4, 5],
    Doubled = [X * 2 || X <- Numbers],
    io:format("Doubled: ~w~n", [Doubled]),
    
    Sum = lists:sum(Numbers),
    io:format("Sum: ~w~n", [Sum]).
